relay-server/src/actors/project_buffer.rs

```diff
@@ -392,7 +401,25 @@ impl BufferService {
     fn extract_envelope(&self, row: SqliteRow) -> Result<ManagedEnvelope, BufferError> {
         let envelope_row: Vec<u8> = row.try_get("envelope")?;
         let envelope_bytes = bytes::Bytes::from(envelope_row);
-        let envelope = Envelope::parse_bytes(envelope_bytes)?;
+        let mut envelope = Envelope::parse_bytes(envelope_bytes)?;
+
+        let received_at: i64 = row.try_get("received_at")?;
+        let received_at = Duration::from_millis(received_at as u64);
+        let elapsed = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .unwrap_or_default()
+            .checked_sub(received_at)
+            .unwrap_or_default();
+
+        let start_time = Instant::now()
+            // Subtract the elapsed time (the time the envelope was kept in the spool) from the current
+            // Instant to get the timestamp when the request started.
+            .checked_sub(elapsed)
+            // If we fail to get the instant from the timestamp, we fallback to `now()`.
+            .unwrap_or_else(Instant::now);
```

Comment:
Why are we shifting the `start_time`? I see you added it to the PR description but I don't know the context on why we need it:

> preserve the current processing time, using start_time field on envelope meta

From the docs of `start_time`:

> The time at which the request started.

If we shift this field, it'll no longer hold and may be quite confusing in the future when we forget a bit about this.

I assume that this hack is the fastest way to keep the total processing time of an envelope correct and ship this fast. If this is the case and we want to move forward with it, let's add a comment to fix this in `start_time`'s docs and create a new follow-up item in the [buffering meta issue](https://github.com/getsentry/team-ingest/issues/53). I'm not blocking the PR because of this, but let's make sure we address this sooner than later.

Reply:
I do not really understand this comment. Could you, please, rephrase it? 

What's happening here is that we preserve the time the envelopes came in when we spool it and then restore it when we deserialize the envelope back for processing. 

What do you mean by hack? At this point this is the working solution, which might be addressed later in the followup refactors, if the underlying structures (e.g. `Envelope` or `ManagedEnvelope`, etc.) are changed.